moveit_ros/moveit_servo/doc/servo_tutorial.md

```diff
@@ -0,0 +1,207 @@
+
+### Inputs/Outputs
+Inputs to `moveit_servo` include
+| Input      | Type | Description     |
+| :---:        |    :----:   |          :---: |
+| Cartesian EEF command  | TwistStamped ROS message | When sent, `Servo` will attempt to move the EEF following the given twist velocity. |
+| Joint command   | JointJog ROS message | When sent, `Servo` will move the joints following the given command |
+| Joint states   | JointState ROS message | Typically from `/joint_states`, the current robot position |
+| Control dimensions   | boolean array | Controls which twist directions commands can be accepted in. Non-controlled directions are set to 0 input. Useful for rejecting input noise (e.g. with 6 DoF mouse) |
+| Drift dimensions   | boolean array | Sets the twist directions that will be ignored in the math. This may result in some motion in the set directions, but can be useful for reducing the task space and avoiding singularities |
+
+`Servo` outputs include:
+| Output      | Type | Description     |
+| :---:        |    :----:   |          :---: |
+| Joint output command  | JointTrajectory OR Float64MultiArray ROS message | Contains the commands to be sent to the manipulator controller |
+| Status   | Int8 ROS message | Encodes the current status of `moveit_servo`. Examples: NO_WARNING, DECELERATE_FOR_COLLISION |
+
+
+### Interface Options
+In ROS2, `moveit_servo` offers 2 interfacing options: a direct C++ API, and as a standalone composable node. 
+
+#### C++ API
+The C++ API is offered as the `Servo` class, defined in `include/moveit_servo/servo.h`. You can add a `Servo` instance to your existing project by:
+
+1) Including `moveit_servo` in your CMakeLists
+2) Including `moveit_servo/servo.h` in your file
+3) Creating a `planning_scene_monitor` (if you don't already have one) and `ServoParameters` and passing both to a new `Servo` object.
+4) Calling `start()` on the `Servo` instance and publishing commands.
+
+See the [C++ interface demo](../src/cpp_interface_demo/servo_cpp_interface_demo.cpp) as an example, and the [Setting Up a New Robot](#Setting-Up-a-New-Robot) section later for details on getting your specific robot up and running.
+
+The C++ API can be visualized with:
+![Alt Text](Images/Servo_Interface.png)
+
+#### Composable Node Interface
+A standalone component is also offered as a `ServoServer`. This allows you include starting a `ServoServer` in your launch file, and interact with `Servo` only through ROS topics and services. See the [servo server demo](../launch/servo_server_demo.launch.py) as an example.
+
+The composable node interface can be visualized with:
+![Alt Text](Images/Servo_Servo_Interface.png)
+
+# Setting Up a New Robot
+
+### Preliminaries
+The bare minimum requirements for running `moveit_servo` on your robot include:
+
+1) A URDF for your robot
+2) A SRDF for your robot
+3) A controller that can accept joint positions or velocities from a ROS topic
```

Comment:
Add this -- we've encountered a few prototype arms where the joint feedback was so bad, Servo didn't work. As far as it could tell, the arm didn't move.

`4) Joint encoders that provide rapid and accurate joint position feedback.`